Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -16,10 +16,27 @@ public class OpenFeatureAPI: GlobalEventPublisher {
     public init() {
     }
 
+    public func setProviderAndWait(provider: FeatureProvider) async {
+        await setProviderAndWait(provider: provider, initialContext: nil)
+    }
+
     public func setProvider(provider: FeatureProvider) {
         self.setProvider(provider: provider, initialContext: nil)
     }
 
+    public func setProviderAndWait(provider: FeatureProvider, initialContext: EvaluationContext?) async {
+        var holder: [AnyCancellable] = []
+        await withCheckedContinuation { continuation in
+            let stateObserver = provider.observe().sink {
```

Comment:
side note for self: Good to test that an event is broadcasted to all the `sink`s when more than one are created